build on musl libc fails due to missing execinfo.h used for stack tracing

On system using musl libc the build fails with:
```
/usr/bin/x86_64-gentoo-linux-musl-g++ -Dkokkoscore_EXPORTS -Icore/src -I/var/tmp/portage/dev-cpp/kokkos-3.0.00/work/kokkos-3.0.00/core/src -I.  -DNDEBUG -O2 -pipe -march=native -fno-common -falign-functions=32:25:16 -fPIC   -fopenmp -MD -MT core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o -MF core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o.d -o core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o -c /var/tmp/portage/dev-cpp/kokkos-3.0.00/work/kokkos-3.0.00/core/src/impl/Kokkos_Stacktrace.cpp
/var/tmp/portage/dev-cpp/kokkos-3.0.00/work/kokkos-3.0.00/core/src/impl/Kokkos_Stacktrace.cpp:6:10: fatal error: execinfo.h: No such file or directory
    6 | #include <execinfo.h>
      |          ^~~~~~~~~~~~
compilation terminated.
```
as musl doesn't have a `execinfo.h`.

See https://bugs.gentoo.org/712748

--- a/core/src/Kokkos_Macros.hpp
+++ b/core/src/Kokkos_Macros.hpp
@@ -601,7 +601,9 @@ static constexpr bool kokkos_omp_on_host() { return false; }
 #if (defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG) ||  \
      defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_PGI)) && \
     !defined(_WIN32) && !defined(__ANDROID__)
+#if __has_include(<execinfo.h>)
 #define KOKKOS_IMPL_ENABLE_STACKTRACE
+#endif
 #define KOKKOS_IMPL_ENABLE_CXXABI
 #endif
 

